numba_dppy/dpnp_glue/dpnp_randomimpl.py

```diff
@@ -0,0 +1,1287 @@
+import numba_dppy.dpnp_glue as dpnp_lowering
+from numba.core.extending import overload, register_jitable
+import numpy as np
+from numba_dppy import dpctl_functions
+import os
+
+
+@register_jitable
+def check_range(low, high):
+    if low >= high:
+        raise ValueError("Low cannot be >= High")
+
+
+@register_jitable
+def common_impl(low, high, res, dpnp_func, PRINT_DEBUG):
+    check_range(low, high)
+
+    sycl_queue = dpctl_functions.get_current_queue()
+    res_usm = dpctl_functions.malloc_shared(res.size * res.itemsize, sycl_queue)
+
+    dpnp_func(res_usm, low, high, res.size)
+
+    dpctl_functions.queue_memcpy(
+        sycl_queue, res.ctypes, res_usm, res.size * res.itemsize
+    )
+    dpctl_functions.free_with_queue(res_usm, sycl_queue)
+
+    dpnp_ext._dummy_liveness_func([res.size])
+
+    if PRINT_DEBUG:
+        print("DPNP implementation")
+
+
+@register_jitable
+def common_impl_0_arg(res, dpnp_func, print_debug):
+    sycl_queue = dpctl_functions.get_current_queue()
+    res_usm = dpctl_functions.malloc_shared(res.size * res.itemsize, sycl_queue)
+
+    dpnp_func(res_usm, res.size)
+
+    dpctl_functions.queue_memcpy(
+        sycl_queue, res.ctypes, res_usm, res.size * res.itemsize
+    )
+    dpctl_functions.free_with_queue(res_usm, sycl_queue)
+
+    dpnp_ext._dummy_liveness_func([res.size])
+
+    if print_debug:
+        print("dpnp implementation")
+
+
+@register_jitable
+def common_impl_1_arg(arg1, res, dpnp_func, print_debug):
+    sycl_queue = dpctl_functions.get_current_queue()
+    res_usm = dpctl_functions.malloc_shared(res.size * res.itemsize, sycl_queue)
+
+    dpnp_func(res_usm, arg1, res.size)
+
+    dpctl_functions.queue_memcpy(
+        sycl_queue, res.ctypes, res_usm, res.size * res.itemsize
+    )
+    dpctl_functions.free_with_queue(res_usm, sycl_queue)
+
+    dpnp_ext._dummy_liveness_func([res.size])
+
+    if print_debug:
+        print("dpnp implementation")
+
+
+@register_jitable
+def common_impl_2_arg(arg1, arg2, res, dpnp_func, print_debug):
+    sycl_queue = dpctl_functions.get_current_queue()
+    res_usm = dpctl_functions.malloc_shared(res.size * res.itemsize, sycl_queue)
+
+    dpnp_func(res_usm, arg1, arg2, res.size)
+
+    dpctl_functions.queue_memcpy(
+        sycl_queue, res.ctypes, res_usm, res.size * res.itemsize
+    )
+    dpctl_functions.free_with_queue(res_usm, sycl_queue)
+
+    dpnp_ext._dummy_liveness_func([res.size])
+
+    if print_debug:
+        print("dpnp implementation")
+
+
+@overload(stubs.dpnp.random)
+@overload(stubs.dpnp.sample)
+@overload(stubs.dpnp.ranf)
+@overload(stubs.dpnp.random_sample)
+def dpnp_random_impl(size):
+    name = "random_sample"
+    dpnp_lowering.ensure_dpnp(name)
+
+    ret_type = types.void
+    """
+    dpnp source:
+    https://github.com/IntelPython/dpnp/blob/0.4.0/dpnp/backend/custom_kernels_random.cpp#L391
+
+    Function declaration:
+    void custom_rng_uniform_c(void* result, long low, long high, size_t size)
+
+    """
```

Comment:
If it is a comment it should be in form of comment.

Reply:
These are comments for future reference. What should be the proper format?

Reply:
I think we should use `#`

Reply:
Unassigned strings also ok.